homeassistant/components/motionmount/config_flow.py

```diff
@@ -0,0 +1,173 @@
+"""Config flow for Vogel's MotionMount."""
+import logging
+from typing import Any
+
+import motionmount
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components import zeroconf
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PORT, CONF_UUID
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.device_registry import format_mac
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+EMPTY_MAC = "00:00:00:00:00:00"
+
+
+# A MotionMount can be in four states:
+# 1. Old CE and old Pro FW -> It doesn't supply any kind of mac
+# 2. Old CE but new Pro FW -> It supplies its mac using DNS-SD, but a read of the mac fails
+# 3. New CE but old Pro FW -> It doesn't supply the mac using DNS-SD but we can read it
+# 4. New CE and new Pro FW -> Both DNS-SD and a read gives us the mac
+# If we can't get the mac, we use DEFAULT_DISCOVERY_UNIQUE_ID as an ID, so we can always configure a single MotionMount. Most households will only have a single MotionMount
+class MotionMountFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a Vogel's MotionMount config flow."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Set up the instance."""
+        self.discovery_info: dict[str, Any] = {}
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initiated by the user."""
+        if user_input is None:
+            return self._show_setup_form()
+
+        info = {}
+        try:
+            info = await self._validate_input(user_input)
+        except ConnectionError:
+            return self.async_abort(reason="cannot_connect")
+        except TimeoutError:
+            return self.async_abort(reason="time_out")
+        except motionmount.NotConnectedError:
+            return self.async_abort(reason="not_connected")
+        except motionmount.MotionMountResponseError:
+            # This is most likely due to missing support for the mac address property
+            # Abort if the handler has config entries already
+            if self._async_current_entries():
+                return self.async_abort(reason="already_configured")
+
+            # Otherwise we try to continue with the generic uid
+            info[CONF_UUID] = config_entries.DEFAULT_DISCOVERY_UNIQUE_ID
+
+        # If the device mac is valid we use it, otherwise we use the default id
+        if info.get(CONF_UUID, EMPTY_MAC) != EMPTY_MAC:
```

Comment:
```suggestion
        if info.get(CONF_UUID) is not None:
```

Reply:
Hi @joostlek, This is not the same: the mac address can be either None or EMPTY_MAC to be invalid in this context.

Reply:
oooh, my mistake!